lib/config.js

```diff
@@ -81,16 +87,44 @@ Config.prototype._loadArchetypeScripts = function (name) {
   // heursitically (hackily) determine if these conditions are true.
   //
   // https://github.com/FormidableLabs/builder/issues/25
-  var pkg;
   try {
-    // Contained.
-    pkg = require(path.join(process.cwd(), "node_modules", name, "package.json"));
+    // Contained in the "usual place"
+    pkgPath = path.join(process.cwd(), "node_modules", name, "package.json");
+    return require(pkgPath);
   } catch (err) {
-    // NPM-installed **and** v3 is a sibling.
-    if (this._isFromNpm && this._isNpmV3) {
-      pkg = require(path.join(process.cwd(), "..", name, "package.json"));
+    /*eslint-disable no-empty*/
+  }
+
+  if (this._isFromNpm) {
+    try {
+      // NPM-installed (sometimes on v2, always on v3)
+      pkgPath = path.join(process.cwd(), "..", name, "package.json");
+      return require(pkgPath);
+    } catch (err) {
+      /*eslint-disable no-empty*/
     }
   }
+
+  try {
+    // Require resolve it
+    var modPath = require.resolve(name);
+    pkgPath = path.join(modPath, "package.json");
+    return require(pkgPath);
```

Comment:
This will break if the archetype has a `main` file (our current archetype doesn't, but I don't think we should break here if an archetype has one for some reason). For example when I try this on `lodash`:

```js
> require("path").join(require.resolve("lodash"), "package.json")
'/Users/brianbeck/Projects/Tests/npm-test/node_modules/lodash/dist/lodash.js/package.json'
```

Also, is there a good reason not to just use this as the one-and-only way? I really believe that needing to crawl up the tree to find the archetype will end up being the common case, not an edge case.